core/src/main/java/org/apache/calcite/rel/rules/ProjectJoinTransposeRule.java

```diff
@@ -59,13 +70,23 @@
    * @param preserveExprCondition Condition for expressions that should be
    *                              preserved in the projection
    */
+  @Deprecated
   public ProjectJoinTransposeRule(
       PushProjector.ExprCondition preserveExprCondition,
       RelBuilderFactory relFactory) {
-    super(
-        operand(Project.class,
-            operand(Join.class, any())),
-        relFactory, null);
+    this(LogicalProject.class, LogicalJoin.class,
```

Comment:
The constructor is deprecated so there is no real need to change its previous behavior.

Reply:
Thanks, I have deleted this constructor.